cpp/src/arrow/util/decimal.h

```diff
@@ -146,6 +146,9 @@ ARROW_EXPORT Decimal128 operator*(const Decimal128& left, const Decimal128& righ
 ARROW_EXPORT Decimal128 operator/(const Decimal128& left, const Decimal128& right);
 ARROW_EXPORT Decimal128 operator%(const Decimal128& left, const Decimal128& right);
 
+Status Rescale(const Decimal128& value, int32_t original_scale, int32_t new_scale,
```

Comment:
I'm renaming this to `RescaleDecimal` since it might be ambiguous in the `arrow` namespace otherwise